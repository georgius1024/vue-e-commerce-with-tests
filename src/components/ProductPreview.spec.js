import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import component from "@/components/ProductPreview.vue";

describe("Product preview", () => {
  const buildWrapper = () => {
    const propsData = {
      product: {
        id: 101,
        title: "product",
        thumbnail: null,
        description: "Lorem ipsum",
        price: 100,
      },
    };
    return mount(component, {
      propsData,
    });
  };
  it("mounts without error",  () => {
    const wrapper = buildWrapper();
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("click on product emits", async () => {
    const wrapper = buildWrapper();
    expect(wrapper.exists).toBeTruthy();
    await wrapper.find('[data-testid="product-preview-101"').trigger("click");
    expect(wrapper.emitted().click).toHaveLength(1);
  });
});
