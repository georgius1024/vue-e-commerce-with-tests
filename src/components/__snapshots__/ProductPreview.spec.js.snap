// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product preview > mounts without error 1`] = `
"<div class=\\"card pointer hover-up\\" style=\\"width: 18rem;\\" data-testid=\\"product-preview-101\\"><img class=\\"card-img-top\\" alt=\\"preview\\">
  <div class=\\"card-body\\">
    <h5 class=\\"card-title\\">product</h5>
    <p class=\\"card-text\\">Lorem ipsum</p><span class=\\"float-end badge rounded-pill fs-5 bg-primary mt-3 me-3\\"> $100</span>
  </div>
</div>"
`;
